ais/src/main/java/au/gov/amsa/ais/AisStaticDataReportPart.java

```diff
@@ -0,0 +1,42 @@
+package au.gov.amsa.ais;
+
+import java.util.Optional;
+
+public enum AisStaticDataReportPart {
+	
+	PART_A(0),
+	PART_B(1);
+	
+	private final int partNumber;
+	
+	private AisStaticDataReportPart(int partNumberCode) {
+		this.partNumber = partNumberCode;
+	}
+	
+	public static Optional<AisStaticDataReportPart> lookup(int partNumberCode) {
+		
+		final AisStaticDataReportPart staticDataReportPart;
+		
+		if(partNumberCode == PART_A.getPartNumber()) {
+			staticDataReportPart = PART_A;
+		} else if (partNumberCode == PART_B.getPartNumber()) {
+			staticDataReportPart = PART_B;
+		} else {
+			staticDataReportPart = null;
+		}
+		
+		return Optional.ofNullable(staticDataReportPart);
+	}
+	
+	public boolean isPartA() {
+		return this==PART_A;
+	}
+	
+	public boolean isPartB() {
+		return this==PART_B;
+	}
+
+	public int getPartNumber() {
+		return partNumber;
+	}
+}
```

Comment:
If really just a binary thing (is A or B) then this class seems unnecessary. I'd vote for simplifying.

Reply:
This was more so for readability as the part number is representative to the part (1=A or 2=B). Happy to change, are you suggesting I use values 1 and 2 where I reference PART_A and PART_B?

Reply:
Just a couple of constants within the class that has the job of parsing as 24 is enough. Keep it simple.

Reply:
Fixed